File: Daily/sbgo/Leetcode766-托普利茨矩阵.go
package sbgo

func isToeplitzMatrix(matrix [][]int) bool {
	m, n := len(matrix), len(matrix[0])
	for i := 1; i < m; i++ {
		for j := 1; j < n; j++ {
			if matrix[i-1][j-1] != matrix[i][j] {
				return false
			}
		}
	}
	return true
}

File: Leetcode/Rust/Leetcode766-托普利茨矩阵.rs
impl Solution {
    pub fn is_toeplitz_matrix(matrix: Vec<Vec<i32>>) -> bool {
        let (m,n) = (matrix.len(),matrix[0].len());
        for i in 1..m{
            for j in 1..n{
                if matrix[i - 1][j - 1] != matrix[i][j]{
                    return false;
                }
            }
        }
        true    
    }
}